.github/workflows/unit-tests.yml

```diff
@@ -1,24 +1,38 @@
-name: TypeScript Unit Tests
-on:
-  pull_request:
-    types:
-      - opened
-      - reopened
-      - ready_for_review
-      - synchronize
+name: Unit tests
+
+on: workflow_call
+permissions: {}
+
+concurrency:
+  group: unit-tests-${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
 jobs:
   test:
+    name: Unit
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-      - name: Setup Node.js
+      - name: Pull repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node
         uses: actions/setup-node@v3
         with:
-          node-version: 18.x
-      - name: Install Dependencies
+          cache: npm
```

Comment:
Removing the node version will leave it up to the [system installed version](https://github.com/actions/setup-node#usage).

If we don't want to be updating this all the time we should either use `node-version-file` and point it at an `.npmrc` file, or set it as `LTS` and have it automatically upgrade once a year.

Reply:
Added back in